url4fdf79ae2c_beckchr_juel_tgz-pJ8-de_odysseus_el_samples_xml_sax_XMELFilterJ8/src/javax/el/BeanELResolver.java

```diff
@@ -0,0 +1,666 @@
+	 * @throws ELException
+	 *             if an exception was thrown while performing the property or variable resolution.
+	 *             The thrown exception must be included as the cause property of this exception, if
+	 *             available.
+	 */
+	@Override
+	public boolean isReadOnly(ELContext context, Object base, Object property) {
+		if (context == null) {
+			throw new NullPointerException();
+		}
+		boolean result = readOnly;
+		if (isResolvable(base)) {
+			result |= toBeanProperty(base, property).isReadOnly();
+			context.setPropertyResolved(true);
+		}
+		return result;
+	}
+
+	/**
+	 * If the base object is not null, attempts to set the value of the given property on this bean.
+	 * If the base is not null, the propertyResolved property of the ELContext object must be set to
+	 * true by this resolver, before returning. If this property is not true after this method is
+	 * called, the caller can safely assume no value was set. If this resolver was constructed in
+	 * read-only mode, this method will always throw PropertyNotWritableException. The provided
+	 * property name will first be coerced to a String. If property is a writable property of base
+	 * (as per the JavaBeans Specification), the setter method is called (passing value). If the
+	 * property exists but does not have a setter, then a PropertyNotFoundException is thrown. If
+	 * the property does not exist, a PropertyNotFoundException is thrown.
+	 * 
+	 * @param context
+	 *            The context of this evaluation.
+	 * @param base
+	 *            The bean to analyze.
+	 * @param property
+	 *            The name of the property to analyze. Will be coerced to a String.
+	 * @param value
+	 *            The value to be associated with the specified key.
+	 * @throws NullPointerException
+	 *             if context is null
+	 * @throws PropertyNotFoundException
+	 *             if base is not null and the specified property does not exist or is not readable.
+	 * @throws PropertyNotWritableException
+	 *             if this resolver was constructed in read-only mode, or if there is no setter for
+	 *             the property
+	 * @throws ELException
+	 *             if an exception was thrown while performing the property or variable resolution.
+	 *             The thrown exception must be included as the cause property of this exception, if
+	 *             available.
+	 */
+	@Override
+	public void setValue(ELContext context, Object base, Object property, Object value) {
+		if (context == null) {
+			throw new NullPointerException();
+		}
+		if (isResolvable(base)) {
+			if (readOnly) {
+				throw new PropertyNotWritableException("resolver is read-only");
+			}
+			Method method = toBeanProperty(base, property).getWriteMethod();
+			if (method == null) {
+				throw new PropertyNotWritableException("Cannot write property: " + property);
+			}
+			try {
+				method.invoke(base, value);
+			} catch (InvocationTargetException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)